Error compiling the example `mesh4all_rpl_dodag`

#### Description
Error compiling the example `mesh4all_rpl_dodag`:
```sh
... examples/mesh4all_rpl_dodag/main.c:28:10: fatal error: shell_commands.h: No such file or directory
   28 | #include "shell_commands.h"
```

#### Steps to reproduce the issue
Change directory to `/examples/mesh4all_rpl_dodag` and try to buid with make.

#### Expected results
Compilation success

#### Actual results
Compilation error

#### Versions
Host OS: Ubuntu 22.04
m4a-firmware: master branch

--- a/examples/mesh4all_rpl_dodag/main.c
+++ b/examples/mesh4all_rpl_dodag/main.c
@@ -25,7 +25,6 @@
 #include <string.h>
 
 #include "shell.h"
-#include "shell_commands.h"
 #include "log.h"
 
 #include "rpl_protocol.h"

--- a/examples/slipdev/main.c
+++ b/examples/slipdev/main.c
@@ -26,7 +26,6 @@
 
 #include "xtimer.h"
 #include "shell.h"
-#include "shell_commands.h"
 #include "msg.h"
 
 #define MAIN_QUEUE_SIZE     (8)

--- a/examples/system_example/main.c
+++ b/examples/system_example/main.c
@@ -25,7 +25,6 @@
 #include <string.h>
 
 #include "shell.h"
-#include "shell_commands.h"
 #include "msg.h"
 #include "log.h"
 #include "board.h"
@@ -34,6 +33,7 @@
 #include "rpl_protocol.h"
 #include "ds18_sensor.h"
 #include "serialization.h"
+#include "uniqueid.h"
 #include "udp_client.h"
 #include "udp_server.h"
 #include "at_client.h"
@@ -66,7 +66,7 @@ void *init_loop(void *args) {
         if (cbor_enconde_message(&payload, buffer, &len_cbor) < 0) {
             printf("error to encode payload with cbor");
         } else {
-            udp_send(&port, ADDRESS_TO_SEND, buffer, &len_cbor);
+            udp_send(port, ADDRESS_TO_SEND, buffer, &len_cbor);
         }
         xtimer_sleep(60);
     }

--- a/firmware/network/radio/Makefile.dep
+++ b/firmware/network/radio/Makefile.dep
@@ -2,4 +2,3 @@ USEMODULE += netdev_default
 USEMODULE += gnrc_ipv6_default
 USEMODULE += gnrc_netif_ieee802154
 USEMODULE += auto_init_gnrc_netif
-USEMODULE += uniqueid

--- a/firmware/sys/uniqueid/Makefile.dep
+++ b/firmware/sys/uniqueid/Makefile.dep
@@ -3,3 +3,9 @@ USEMODULE += ipv6_addr
 USEMODULE += netdev_default
 USEMODULE += gnrc_ipv6_default
 USEMODULE += auto_init_gnrc_netif
+
+ifneq (,$(filter m4a-24g m4a-mb,$(BOARD)))
+    USEMODULE += radio
+else
+    USEMODULE += periph_hwrng
+endif

--- a/firmware/sys/uniqueid/uniqueid.c
+++ b/firmware/sys/uniqueid/uniqueid.c
@@ -17,6 +17,7 @@
  * @brief This function gets ipv6 address (mode: static (default), random, manual)
  *
  * @author  RocioRojas <rociorojas391@gmail.com>
+ * @author  eduazocar <eduazocarv@gmail.com>
  */
 
 #include <stdio.h>
@@ -26,36 +27,11 @@
 #include "random.h"
 #include "net/netdev/ieee802154.h"
 #include "net/gnrc.h"
-
-#ifdef CONFIG_MODE_RANDOM
-static uint8_t radio_devices[] = {NETDEV_AT86RF215, NETDEV_AT86RF2XX, NETDEV_CC2538};
-
-#ifdef CONFIG_MODE_SUB_24GHZ
-int8_t subtract_to_interface_radio = 1;
+#if MODULE_AT86RF2XX || MODULE_AT86RF215
+#include "radio.h"
 #else
-int8_t subtract_to_interface_radio = 0;
+#include "periph/hwrng.h"
 #endif
-
-int8_t get_ieee802154_iface_radio(void) {
-    int max_ifaces = gnrc_netif_numof();
-    if (max_ifaces > 0) {
-        gnrc_netif_t *iface;
-        for (uint8_t i = 0; i < ARRAY_SIZE(radio_devices); i++) {
-            iface = gnrc_netif_get_by_type(radio_devices[i], NETDEV_INDEX_ANY);
-            if (iface != NULL) {
-                break;
-            }
-        }
-        if (iface != NULL) {
-            return iface->pid - subtract_to_interface_radio;
-        } else {
-            return -1;
-        }
-    }
-    return -1;
-}
-#endif
-
 void subnet_to_ipv6(ipv6_addr_t *addr) {
 
 #ifdef CONFIG_MODE_STATIC
@@ -74,19 +50,20 @@ void subnet_to_ipv6(ipv6_addr_t *addr) {
     ipv6_addr_t header = {
         .u8 = {0},
     };
-    uint32_t test = 0;
-    int index = get_ieee802154_iface_radio();
-    netif_t *iface = netif_get_by_id(index);
+    union random_buff random_number;
     ipv6_addr_from_str(&header, CONFIG_HEADER_ADDRESS_ID);
     memcpy((char *)addr->u8, (char *)header.u8, 4);
-    union random_buff random_number;
-    netif_get_opt(iface, NETOPT_RANDOM, 0, &test, sizeof(test));
-    int seed = test;
-    random_init(seed);
+#if (MODULE_AT86RF2XX || MODULE_AT86RF215)
+    int index = get_ieee802154_iface();
+    netif_t *iface = netif_get_by_id(index);
+    netif_get_opt(iface, NETOPT_RANDOM, 0, &random_number, sizeof(random_number));
+#else
+    hwrng_init();
+    hwrng_read(&random_number, sizeof(random_number));
+#endif
+    random_init(random_number.u32);
     random_number.u32 = random_uint32();
-    (void)random_number.u32;
     strncat((char *)addr->u8, (char *)random_number.u8, 4);
-
 #endif
 
 #ifdef CONFIG_MODE_MANUAL

--- a/tests/uniqueid/main.c
+++ b/tests/uniqueid/main.c
@@ -19,12 +19,15 @@
  *
  * @author  xkevin190 <kevinvelasco193@gmail.com>
  * @author  RocioRojas <rociorojas391@gmail.com>
+ * @author  eduazocar <eduazocarv@gmail.com>
  */
+#include <stdio.h>
 #include <string.h>
 #include <errno.h>
 
 #include "embUnit.h"
 #include "uniqueid.h"
+#include "unique_random.h"
 
 void get_unique_from_mac(ipv6_addr_t *output) {
     ipv6_addr_t header = {
@@ -68,6 +71,8 @@ void test_get_ipv6Address(void) {
     printf("\nSecond random IPv6\n");
     ipv6_addr_print(&output2);
     TEST_ASSERT_EQUAL_INT(0, ipv6_addr_equal(&output1, &output2));
+    printf("\n\nBinary data printed\n");
+    random_generator(64);
 #endif
 }
 

--- a/tests/uniqueid/random_gen.c
+++ b/tests/uniqueid/random_gen.c
@@ -1,0 +1,65 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @author  eduazocar <eduazocarv@gmail.com>
+ */
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include "embUnit.h"
+#include "uniqueid.h"
+#include "random.h"
+#include "unique_random.h"
+
+#if (MODULE_AT86RF2XX || MODULE_AT86RF215)
+#include "radio.h"
+#endif
+#define BIT_ON_MASK (0x01)
+
+void random_generator(uint8_t bit_ref) {
+
+#if (MODULE_AT86RF2XX || MODULE_AT86RF215)
+    int index = get_ieee802154_iface();
+    netif_t *iface = netif_get_by_id(index);
+#endif
+    uint8_t avg_bit32 = bit_ref / 32;
+    if (bit_ref % 32 != 0) {
+        avg_bit32++;
+    }
+    uint32_t random_byte_block = 0;
+    for (uint8_t i = 0; i < bit_ref; i++) {
+        printf("ref %u", (uint8_t)(i + 1));
+        for (uint8_t j = 0; j < avg_bit32; j++) {
+#if (MODULE_AT86RF2XX || MODULE_AT86RF215)
+            netif_get_opt(iface, NETOPT_RANDOM, 0, &random_byte_block, sizeof(uint32_t));
+#endif
+            random_init(random_byte_block);
+            random_byte_block = random_uint32();
+            printf(" Block: %d\t%" PRIx32 " \t", j, random_byte_block);
+            print_binary(random_byte_block, bit_ref / avg_bit32);
+            printf("\t");
+        }
+        printf("\n");
+    }
+}
+
+void print_binary(uint32_t val, uint8_t len) {
+    for (int16_t i = len - 1; i >= 0; i--) {
+        printf("%u", (uint8_t)((val >> i) & BIT_ON_MASK));
+    }
+}

--- a/tests/uniqueid/unique_random.h
+++ b/tests/uniqueid/unique_random.h
@@ -1,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief   Randomizer to obtain values hex in binary
+ *
+ * @author  eduazocar <eduazocarv@gmail.com>
+ */
+#ifndef UNIQUE_RANDOM_H
+#define UNIQUE_RANDOM_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Random Generator to gets a block of random values
+ *
+ * @param bit_ref this is a reference to create a matrix of the reads
+ * e.g bit_ref = 9 -> 9x9 (This mean that could obtain 9 measures of 9
+ * random bits generated )
+ */
+void random_generator(uint8_t bit_ref);
+
+/**
+ * @brief   Printer to random values
+ *
+ * @param val Any input value in the range of 32 bits
+ * @param len length of the bits int @p val
+ */
+void print_binary(uint32_t val, uint8_t len);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* UNIQUE_RANDOM_H */
+/** @} */

